src/parse.cpp

```diff
@@ -541,6 +541,12 @@ expression* parser::parse_async_expression_only(token async_token) {
         this->skip();
         continue;
       }
+      if (this->peek().type == token_type::kw_await) {
+        this->error_reporter_->report(
+            error_await_parameter_disallowed_in_async_arrow_function{
+                .await_token = this->peek().span(),
+            });
```

Comment:
Blocking: We should only emit an error if we know this is an arrow function. Here's a legal example of `async(await` which is not an arrow function:
```js
var async = console.log;
var await = 'taco';
async(await);
```